libraries/AP_Eigen/AP_Algebra.h

```diff
@@ -0,0 +1,165 @@
+#ifndef AP_EXTH_H
+#define AP_EXTH_H
+
+#include <limits>
+#include <type_traits>
+#include <float.h>
+
+#undef min
+#undef max
+
+/*
+ * @brief: Safe way to convert floating point numbers to integers
+ * @authors: Daniel Frenzel <dgdanielf@gmail.com>
+ */
+template <class IntType, class FloatType>
+static inline IntType safeFloatToInt(const FloatType &num) {
+    //check if float fits into integer
+    if(std::numeric_limits<IntType>::digits > std::numeric_limits<FloatType>::digits) {
```

Comment:
We are trying very hard to keep the coding style consistent in the project. It would be really nice if new code like this would follow it. Here is the page with examples: http://dev.ardupilot.com/wiki/style-guide/

The mistakes I'm seeing throughout the PR are in general related to spaces. It should be:
  - `if (std::...)`, not `if(std::..)`;
  - `for (std::...)`, not `for(std::...)`;
  - `std::numeric_limits<IntType>::digits>`, not `std::numeric_limits<IntType>::digits >`;
  - functions should have their braces in a new line, not in the same line. This applies **only** to functions.
  - function and variable names should not use camelCase, use underscores